src/test/template-shadowroot.ts

```diff
@@ -0,0 +1,307 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {hasNativeDeclarativeShadowRoots, hydrateShadowRoots} from '../template-shadowroot.js';
+
+const elementLog: Array<string|null> = [];
+
+class TestLogElement extends HTMLElement {
+  constructor() {
+    super();
+    const label = this.getAttribute('label');
+    elementLog.push(label);
+  }
+}
+customElements.define('test-log', TestLogElement);
+
+// When we serialize
+function assertSerializesAs(actual: Element, expectedSerialization: string) {
+  let actualSerialization = getSerializableTree(actual).innerHTML;
+  actualSerialization = actualSerialization.replace(/\n\s+/g, '');
+  expectedSerialization = expectedSerialization.replace(/\n\s+/g, '');
+  expect(actualSerialization).toEqual(expectedSerialization);
```

Comment:
Can we please use Chai's assertion API?

Reply:
y tho

One nice part about this API is that it makes it very clear which is the actual and which is the expected. With chai we often get this backwards, making errors harder to read